### PR TITLE
fix(connectors): fixes MGDCTRS-860, fixed admin API connector delete

### PR DESCRIPTION
**Pull request by dhirajsb**
## Description
Fixes MGDCTRS-860, fixes admin API connector delete

## Verification Steps
Added integration test should pass. 

## Checklist (Definition of Done)
- [X] All acceptance criteria specified in JIRA have been completed
- [X] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
~~- [ ] Documentation added for the feature~~
- [ ] CI and all relevant tests are passing
- [ ] Code Review completed
~~- [ ] Verified independently by reviewer~~
~~- [ ] Required metrics/dashboards/alerts have been added (or PR created).~~
~~- [ ] Required Standard Operating Procedure (SOP) is added.~~
~~- [ ] JIRA has created for changes required on the client side~~

### CODE REVIEW

File: internal/connector/test/integration/features/connector-agent-api.feature [hunk 1]
```diff
@@ -1160,9 +1160,72 @@ Feature: connector agent API
...
+    # force delete connector using admin API
+    When I DELETE path "/v1/admin/kafka_connectors/${forced_connector_id}?force=true"
+    Then the response code should be 204
+
+    # assigned connector and deployment should get deleted immediately
+    When I GET path "/v1/admin/kafka_connectors/${forced_connector_id}"
+    Then the response code should be 410
+    When I GET path "/v1/admin/kafka_connector_clusters/${connector_cluster_id}/deployments"
+    Then the response code should be 200
+    And the ".total" selection from the response should match "1"
```
Comment (lburgazzoli) [inline internal/connector/test/integration/features/connector-agent-api.feature:1227]: We should also check that the deployment related to the `forced_connector_id` is not more present in the response

Comment (lburgazzoli) [reply internal/connector/test/integration/features/connector-agent-api.feature:1227]: Do we also need a test to verify the secrets are wiped ?

**Comment by lburgazzoli** [comment]
LGTM, added a note about some additional tests/checks

File: internal/connector/test/integration/features/connector-agent-api.feature (hunk 1 above)
Comment (dhirajsb) [reply internal/connector/test/integration/features/connector-agent-api.feature:1227]: yeah, that's a good additional check.

Comment (dhirajsb) [reply internal/connector/test/integration/features/connector-agent-api.feature:1227]: Secrets deletion verification is done in another connector delete test already, and force delete now reuses the existing Delete() func. Do we really need to verify that again?

Comment (lburgazzoli) [reply internal/connector/test/integration/features/connector-agent-api.feature:1227]: > Secrets deletion verification is done in another connector delete test already, and force delete now reuses the existing Delete() func. Do we really need to verify that again?

if it is done in another test, then it is fine.
Was unsure about the scope of this fix

Comment (dhirajsb) [reply internal/connector/test/integration/features/connector-agent-api.feature:1227]: The admin soft delete and force delete implementation now reuses existing delete functions from the public API, just without the authz checks for connector ownership. The previous implementation duplicated _some_ of that code, hence the issue.

**Review by valdar — APPROVED** [review]

**Pull request merged**